### PR TITLE
Forward port ctr status on Thin Controller.

### PR DESCRIPTION
**Pull request by corneil**
Added ThinTaskExecution to store ctrStatus.
Update Controller and Service to populate the ctr status.

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/impl/DefaultDataFlowTaskExecutionQueryDao.java [hunk 1]
```diff
@@ -509,4 +521,25 @@ private List<String> getTaskArguments(long taskExecutionId) {
...
+				.map(Object::toString)
+				.collect(Collectors.joining(","));
+		String sql = FIND_CTR_STATUS.replace(":taskExecutionIds", ids);
+		jdbcTemplate.query(sql, rs -> {
+			Long id = rs.getLong("TASK_EXECUTION_ID");
+			String ctrStatus = rs.getString("CTR_STATUS");
+			logger.debug("populateCtrStatus:{}={}", id, ctrStatus);
+			ThinTaskExecution execution = taskExecutionMap.get(id);
+			Assert.notNull(execution, "Expected TaskExecution for " + id + " from " + ids);
+			updated.incrementAndGet();
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/impl/DefaultDataFlowTaskExecutionQueryDao.java:541]: Not sure why we need the AtomicInteger here.   I don't see it used.

**Review by cppwfs — CHANGES_REQUESTED** [review]
LGTM just a couple of Nitpicks.

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ThinTaskExecution.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+package org.springframework.cloud.dataflow.core;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.cloud.task.repository.TaskExecution;
+
+/**
+ * Provide for ctr status
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ThinTaskExecution.java:24]: Yes

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/DataflowTaskExecutionQueryDao.java [hunk 3]
```diff
@@ -166,4 +167,5 @@ public interface DataflowTaskExecutionQueryDao {
 
 	TaskExecution geTaskExecutionByExecutionId(String executionId, String taskName);
 
+	void populateCtrStatus(Collection<ThinTaskExecution> aggregateTaskExecutions);
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/DataflowTaskExecutionQueryDao.java:170]: definitely

[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/DataflowTaskExecutionQueryDao.java:170]: updated all usages

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/DataflowTaskExplorer.java [hunk 4]
```diff
@@ -175,4 +176,10 @@ public interface DataflowTaskExplorer {
 	 * @see #getLatestTaskExecutionsByTaskNames(String...)
 	 */
 	TaskExecution getLatestTaskExecutionForTaskName(String taskName);
+
+	/**
+	 * Populate CTR status for all tasks
+	 * @param aggregateTaskExecutions
+	 */
+	void populateCtrStatus(Collection<ThinTaskExecution> aggregateTaskExecutions);
```
[the remark this answers is not in the document]
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/DataflowTaskExplorer.java:184]: updated all usages

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/impl/DefaultDataFlowTaskExecutionQueryDao.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/task/impl/DefaultDataFlowTaskExecutionQueryDao.java:541]: it was logged previously. removed now.

**Review by cppwfs — APPROVED** [review]
LGTM.   Thanks for the great work!

**Pull request merged**